core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java

```diff
@@ -138,8 +143,19 @@ public boolean accept(CarbonFile file) {
               if (filesToBeDeleted.length == 0) {
                 status = true;
               } else {
-
                 for (CarbonFile eachFile : filesToBeDeleted) {
+                  // If the file to be deleted is a carbondata file, index file, index merge file
+                  // or a delta file, copy that file to the trash folder.
+                  if ((eachFile.getName().endsWith(CarbonCommonConstants.FACT_FILE_EXT) ||
```

Comment:
better to check by LoadMetadataDetails level, not by file level

Reply:
coping segment wise in the case of normal table, in the case of partition flow have kept it file by file.

Reply:
mention this in comment, why we need to do like this

Reply:
changed, it's not needed anymore